lib/services/profiles-api-service.js

```diff
@@ -75,32 +75,29 @@ function profileApiServiceFactory(
         return _.flattenDeep([macs]);
     };
 
-    ProfileApiService.prototype.setLookup = function(req, res) {
-        var query = req.swagger ? req.swagger.query : req.query;
+    ProfileApiService.prototype.getRequestMacAddress = function(query, ipAddress) {
```

Comment:
appreciate extracting this function, but it's better to refine the function name and it's parameter, maybe getMacAddressInRequest or other name is more clear. and the parameter ipAddress here is specific from 'res'